### ISSUE TITLE
Problem to start weave (Error response from daemon ... Cannot find child for /weave)

### ISSUE THREAD

**Issue by fikovnik** [weaveworks/weave#835]
I'm having a problem to start weave or actually it does start but only on the second time. The images are all pulled.

Here is what happens:

```
$ docker images | grep weave
weaveworks/weaveexec     0.11.1              9b6edd9b1974        45 hours ago        43.57 MB
weaveworks/weavedns      0.11.1              6a7466bbb449        45 hours ago        9.909 MB
weaveworks/weave         0.11.1              cc132cb20eb8        45 hours ago        14.04 MB

$ weave launch -iprange 10.1.1.0/24 -initpeercount 3
2015/06/03 12:20:52 Error response from daemon: Cannot start container 9e270314944a92763111fa7d07622d840ff7baee7029fec3d5414a7a8835857b: Cannot find child for /weave

$ weave launch -iprange 10.1.1.0/24 -initpeercount 3
e7ad762d089c06d8d9cd2a56e5e3414f383dbfa721e91316aa4659d9875b97c2
```

The same is for weave DNS and weave proxy.

Docker info:
```sh
$ docker info
Containers: 3
Images: 48
Storage Driver: devicemapper
 Pool Name: docker-253:1-134574-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.268 GB
 Data Space Total: 107.4 GB
 Data Space Available: 18.11 GB
 Metadata Space Used: 2.572 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Kernel Version: 3.13.0-53-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 4
Total Memory: 7.798 GiB
Name: datalyse-2
ID: FZZQ:FT33:RH2B:GHPH:CBW2:QBXC:ND64:VZ43:YPVL:L6JX:RK3W:OKML
Debug mode (server): true
Debug mode (client): false
Fds: 20
Goroutines: 31
System Time: Wed Jun  3 12:24:04 UTC 2015
EventsListeners: 1
Init SHA1: 7f9c6798b022e64f04d2aff8c75cbf38a2779493
Init Path: /usr/bin/docker
Docker Root Dir: /var/lib/docker
WARNING: No swap limit support
```

**Comment by rade** [comment]
what does `docker version` say?

**Comment by fikovnik** [comment]
That was quick :-)

Here is the log: https://gist.github.com/fikovnik/9f40f9f6c9acdd79a060

**Comment by rade** [comment]
Did the output in the original post get truncated? I am asking because the logs say "Cannot find child for /weave**proxy**"

**Comment by rade** [comment]
I wonder whether you are running into https://github.com/docker/compose/issues/1304, which is likely a docker issue, even though it's filed against 'compose'.

**Comment by fikovnik** [comment]
Sorry, my bad. Actually, the initial post was about the weave but the same is for weavedns and weaveproxy. Once you asked for the log I did run the weaveproxy since I have already run weave and the second+ times it seems to work. Basically every time I restart docker I have to run all commands twice to work. I don't know if is any related, but I use Ansible (with the module command).

Regarding to docker/compose#1304 I had a look at it yesterday, but so far it is only weave that exhibit this behavior. Other containers seem to be happy.

**Comment by rade** [comment]
Try resetting docker. Something like

````
service docker stop
rm -rf /var/lib/docker
service docker start
````
Then pull the weave images explicitly with `weave --local setup`.

And then try `weave launch`.

**Comment by fikovnik** [comment]
Interesting, but so far so good:

```
root@datalyse-2:~# service docker stop
docker stop/waiting

root@datalyse-2:~# rm -fr /var/lib/docker

root@datalyse-2:~# service docker start
docker start/running, process 1399

root@datalyse-2:~# docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

root@datalyse-2:~# weave --local setup
0.11.1: Pulling from weaveworks/weave
c0e52ded2709: Pull complete 
652858803e62: Pull complete 
46f29bdd60d8: Pull complete 
cc132cb20eb8: Already exists 
Digest: sha256:5776d24e5ee994a6aeb554fdd01e98c36b968bea2faca32fce53b35e1f5e5d2a
Status: Downloaded newer image for weaveworks/weave:0.11.1
0.11.1: Pulling from weaveworks/weavedns
097a63cf1f9e: Pull complete 
dd90940c8e60: Pull complete 
6a7466bbb449: Already exists 
c0e52ded2709: Already exists 
652858803e62: Already exists 
Digest: sha256:04c19243319e229cd8f5df7a6069f36507e18684c49dde39eb9ddd4db01ea0f7
Status: Downloaded newer image for weaveworks/weavedns:0.11.1
0.11.1: Pulling from weaveworks/weaveexec
511136ea3c5a: Pull complete 
3adc3de69ee5: Pull complete 
17b3579b6190: Pull complete 
43958670dd26: Pull complete 
862b7ec3f268: Pull complete 
5f4ee6e4d836: Pull complete 
a8d0e33e7e37: Pull complete 
5ac2dadf765b: Pull complete 
5e1427df3eec: Pull complete 
466d01606d66: Pull complete 
7fdd13bedf6b: Pull complete 
9b6edd9b1974: Already exists 
Digest: sha256:02f44520a5a1c5a424a24ccca73dba0a5ff02612a4385123e9f8e6c0032cc50e
Status: Downloaded newer image for weaveworks/weaveexec:0.11.1

root@datalyse-2:~# weave launch -iprange 10.1.1.0/24 -initpeercount 3
4702fae2f1b7dd58f15eb4a4b3f3045b880c1c8db2953e7502bdb415152c2547
```

I will try a bit more and then hopefully close the ticket!

**Comment by rade** [comment]
Well, I am sure there is *some* problem there. With docker. So if you can find any way to reproduce the issue from a clean slate, then I'm sure the docker devs would be delighted to see a bug report for it.

**Comment by fikovnik** [comment]
True, but the reproducibility is indeed the main problem. Anyway, I just checked with all the nodes after restart and it seems ok.

Thanks a lot for quick support!

**Comment by erik-landvall** [comment]
Landed on this post from google after a similar issue. Just wanted to point out that I solved it much the same way by doing `docker ps -a` that showed me some other containers not running and not exited. I removed them by id, no name was attached: `docker rm ######` and then started the container again. It worked, but I'm ofc still confused to what happened. 

In my case, I had a memory issue that caused the service in the container to crash originally. Idk if it's relative, other containers was still running fine as well (but using 90+ % processor power because of an endless-loop-bug caused by the sibling service being down), but still throw the same error as explained above.
